rust/flatbuffers/src/vector.rs

```diff
@@ -147,3 +153,116 @@ impl<'a, T: Follow<'a> + 'a> Follow<'a> for Vector<'a, T> {
         Vector::new(buf, loc)
     }
 }
+
+#[derive(Debug)]
+pub struct VectorIter<'a, T: 'a> {
+    buf: &'a [u8],
+    loc: usize,
+    remaining: usize,
+    phantom: PhantomData<T>,
+}
+
+impl<'a, T: 'a> VectorIter<'a, T> {
+    #[inline]
+    pub fn new(inner: Vector<'a, T>) -> Self {
+        VectorIter {
+            buf: inner.0,
+            loc: inner.1 + SIZE_UOFFSET,
```

Comment:
Please include a comment explaining what that the `+ SIZE_UOFFSET` is for.

Reply:
Done.